007_team_5_agent_design/007_team_5_agent_design.tex

```diff
@@ -1 +1,532 @@
+
+\section{Social Memory}\label{sec:team5-memory}
+The agent stores the information gained from communication in its `memory': the information stored by this data structure is shown in Table \ref{tab:team5-memory}.
+
+\begin{table}
+    \centering
+    \begin{tabular}%
+        {| >{\raggedleft\arraybackslash}p{0.25\linewidth} | %
+        >{\raggedright\arraybackslash}p{0.65\linewidth} | %
+        }
+        \hline
+        Agent ID & An agent's unique ID\\
+        \hline
+        Agent HP & The agent's last known HP level\\
+        \hline
+        Food Taken & The amount of food the agent last took\\
+        % \hline
+        % Intended Food Intake & The amount of food the agent intends to take\\
+        \hline
+        Favour & The Team 5 agent's opinion of this agent\\
+        \hline
+        Days Since Last Seen & The number of days since the last message from this agent\\
+        \hline
+    \end{tabular}
+    \caption{Team 5 Agent Memory}
+    \label{tab:team5-memory}
+\end{table}
+
+This memory allows the agent to construct a social network of agents in the tower and evaluate others based on these parameters. This consequently leads to the idea of favour, a metric the Team 5 agent uses to judge other agents. If more than five days have passed since the agent's last interaction with another agent, then the knowledge gained of that agent is reset, with the favour value being kept. This prevents the agent from using out-of-date information in its calculation while still allowing the opinion formed of the agent to be retained indefinitely.
+
+\subsection*{Favour}\label{sec:team5-favour}
+The favour metric quantifies the agent's opinion of others in the tower. It is bounded between 0 and 10, and in a single tick can either increase by 1, decrease by 3, or stay the same:
+
+\begin{align*}
+    \texttt{hpScoreOther} &= -1 \times \frac{\texttt{otherHP}^{1.7} \times \texttt{foodTaken}^{1.3}}{\texttt{maxHP}^3}\\
+    \texttt{hpScoreSelf} &= \frac{\texttt{ownHP}^{1.7}\times\texttt{ownAttemptFood}^{1.3}}{\texttt{maxHP}^3}\\
+    \texttt{judgement} &= 100 \times (\texttt{hpScoreOther} + \texttt{hpScoreSelf})\\
+    \texttt{unboundedFavour} &= \begin{cases}
+        \texttt{originalFavour} + 1, \quad\hfill \texttt{judgement} &> 0.075\\
+        \texttt{originalFavour} + \max{\left(\frac{\texttt{judgement}}{2}, -3\right)}, \quad\hfill\texttt{judgement} &< -2
+    \end{cases}\\
+    \texttt{favour} &= \begin{cases}
+        0, \quad\hfill \texttt{unboundedFavour} &< 0\\
+        \texttt{unboundedFavour}, \quad 0 \leq \texttt{unboundedFavour} &\leq 10\\
+        10, \quad\hfill \texttt{unboundedFavour} &> 10
+    \end{cases}\\
+\end{align*}
+
+We have designed the favour calculation in such a way that means positive changes in favour occur slowly while negative changes in favour can occur quickly; this is to reflect human behaviour regarding trust, which tends to be lost more quickly than it is gained.
```

Comment:
Just as a suggestion, if there is a paper you can quote on this I think Pitt would like it

Reply:
I'm unable to find a paper to back up this claim, probably best that we remove it in that case.
```suggestion
We have designed the favour calculation in such a way that means positive changes in favour occur slowly while negative changes in favour can occur quickly; this was done to design an agent which builds trust slowly overtime but will distrust at a rate reflective of the severity of another's actions.
```